File: android-beans/src/main/java/rocks/frieler/android/beans/BeanInstantiationException.java
package rocks.frieler.android.beans;

/**
 * {@link RuntimeException} thrown, when beans could not be instantiated.
 *
 * @author Christopher Frieler
 */
public class BeanInstantiationException extends RuntimeException {
    public BeanInstantiationException(String message) {
        super(message);
    }

    public BeanInstantiationException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: android-beans/src/main/kotlin/rocks/frieler/android/beans/BeanInstantiationException.kt
package rocks.frieler.android.beans

/**
 * [RuntimeException] thrown, when beans could not be instantiated.
 *
 * @author Christopher Frieler
 */
class BeanInstantiationException : RuntimeException {
	constructor(message: String?) : super(message)
	constructor(message: String?, cause: Throwable?) : super(message, cause)
}
